packages/extractor-helpers/it/job-helper.test.js

```diff
@@ -0,0 +1,227 @@
+    await settingsStore.putSettings({
+      sourceId,
+    });
+    sources.push(sourceId);
+  });
+  afterEach(async () => settingsStore.deleteSettings(sourceId));
+
+  describe('full', () => {
+    beforeEach(async () => {
+      jobId = await jobHelper.start(sourceId, JobHelper.JOB_TYPE.FULL);
+    });
+
+    it('can complete', async () => {
+      await jobHelper.complete(jobId, sourceId, 'test-cursor');
+
+      assert.ok(jobHelper.shouldStop(jobId));
+
+      const currentSettings = await settingsStore.getSettings(sourceId);
+      assert.ok(currentSettings.currentJobStarted);
+      delete currentSettings.currentJobStarted;
+      assert.ok(currentSettings.lastJobDone);
+      delete currentSettings.lastJobDone;
+      assert.deepStrictEqual(currentSettings, {
+        currentJobStatus: JobHelper.JOB_STATUS.COMPLETE,
+        cursor: 'test-cursor',
+        sourceId,
+      });
+    });
+
+    it('can start another full job after completing', async () => {
+      await jobHelper.complete(jobId, sourceId, 'test-cursor');
+
+      const currentJobId = await jobHelper.start(
+        sourceId,
+        JobHelper.JOB_TYPE.FULL,
+      );
+      assert.ok(currentJobId);
+
+      const currentSettings = await settingsStore.getSettings(sourceId);
+      assert.ok(currentSettings.currentJobStarted);
+      delete currentSettings.currentJobStarted;
+      assert.ok(currentSettings.lastJobDone);
+      delete currentSettings.lastJobDone;
+      assert.deepStrictEqual(currentSettings, {
+        currentJobId,
+        currentJobStatus: JobHelper.JOB_STATUS.RUNNING,
+        sourceId,
+        cursor: 'test-cursor',
+      });
+    });
+
+    it('can stop', async () => {
+      await jobHelper.stop(jobId, sourceId);
+
+      assert.ok(jobHelper.shouldStop(jobId));
+
+      const currentSettings = await settingsStore.getSettings(sourceId);
+      assert.ok(currentSettings.currentJobStarted);
+      delete currentSettings.currentJobStarted;
+      assert.deepStrictEqual(currentSettings, {
+        currentJobStatus: JobHelper.JOB_STATUS.STOPPED,
+        sourceId,
+      });
+    });
+
+    it('can start another full job after stopping', async () => {
+      await jobHelper.stop(jobId, sourceId);
+
+      const currentJobId = await jobHelper.start(
+        sourceId,
+        JobHelper.JOB_TYPE.FULL,
+      );
+      assert.ok(currentJobId);
+
+      const currentSettings = await settingsStore.getSettings(sourceId);
+      assert.ok(currentSettings.currentJobStarted);
+      delete currentSettings.currentJobStarted;
+      assert.deepStrictEqual(currentSettings, {
+        currentJobId,
+        currentJobStatus: JobHelper.JOB_STATUS.RUNNING,
+        sourceId,
+      });
+    });
+
+    it('cannot start another full job', async () => {
+      await assert.rejects(
+        () => jobHelper.start(sourceId, JobHelper.JOB_TYPE.FULL),
+        { status: 409 },
+      );
+    });
+
+    it('can start start update job', async () => {
```

Comment:
```suggestion
    it('can start update job', async () => {
```

Reply:
Yes, the idea is that one update and one full job can execute at a time. That way full jobs don't block update jobs and the reverse.